City-of-Helsinki/drupal-hdbt-admin
# UHF-4815: Update Gin version

# Update Gin theme and Gin toolbar [UHF-4815](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4815)

Gin theme got new beta version during spring and it had multiple design changes. These changed didn't apply out of the box to HDBT Admin theme.

## What was done
* Updated Gin to latest version
* Updated HDBT Admin block and added update hook to automatically fix the region positions for the blocks.
* Updated admin theme regions to match Gin theme.
* Updated templates based on Gin theme new design.
* Updated @sideway/address and glob node modules.
* Removed obsolete TPR edit form javascript ginifier.
* Removed unnecessary Gin sidebar simulation from TPR unit and TPR service. 
* Refactored styles for Gin theme design changes.
* Added hook_gin_content_form_routes to register routes to apply Gin’s content edit form layout on TPR unit and taxonomy terms.
* Added styleslinter to unify the styling conventions with HDBT theme and fixed most obvious violations based on lint rules. 
* Added hel.fi specific title, description and field visibility alterations to all entities instead of just nodes.
* Added divider for the mobile sidebar.
* Added Gin sidebar to taxonomy term forms.
* Added "hdbt-admin" class to body.
* Added fixed base styles for HDBT Admin theme and altered entity from save button label to exclude "(this translation)".
* Added missing configurations and deleted obsolete local tasks block.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the HDBT Admin theme
    * `composer require drupal/hdbt_admin:dev-UHF-4815_gin_update --with-all-dependencies && composer require drupal/hdbt:dev-UHF-4815_gin_update && composer require drupal/helfi_platform_config:dev-UHF-4815_gin_update --with-all-dependencies`
* Run `make drush-updb drush-cr`

## How to test
* Log in to the site and explore admin theme. Look for broken and inconsistent styles. 
Check at least:
  * Node add / edit / revision / preview 
  * TPR Unit, TPR Service, TPR Service channel, TPR Errand service
  * Taxonomy term add / edit form
  * Paragraphs  
     * Test all which uses ajax, like media entities, hero, content links...
  * Menus and menu item editing works
  * Content and TPR Integration views

## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/309
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/317
